lsp-mode.el

```diff
@@ -2207,6 +2226,72 @@ Applies on type formatting."
                            (lambda (edits)
                              (when (= tick (buffer-chars-modified-tick)) (lsp--apply-text-edits edits))))))))
 
+(defun lsp--set-document-link-timer ()
+  (when (and lsp-enable-links (lsp--capability "documentLinkProvider"))
+    (setq-local lsp--links-idle-timer (run-with-idle-timer
+                                       lsp-links-check-internal nil
+                                       #'lsp--update-document-links
+                                       (current-buffer)))))
+
+(defun lsp--cancel-document-link-timer ()
+  (when lsp--links-idle-timer
+    (cancel-timer lsp--links-idle-timer)
+    (setq-local lsp--links-idle-timer nil)))
+
+(defun lsp--update-document-links (&optional buffer)
+  (when (or (not buffer) (and (eq (current-buffer) buffer)
+                              (buffer-live-p buffer)))
+    (cl-assert (lsp--capability "documentLinkProvider"))
+    (let ((buffer (current-buffer)))
+      (lsp-request-async "textDocument/documentLink"
+                         `(:textDocument ,(lsp--text-document-identifier))
+                         (lambda (links)
+                           (seq-do (lambda (overlay)
+                                     (delete-overlay overlay))
+                                   lsp--link-overlays)
+                           (seq-do
+                            (lambda (link)
+                              (with-current-buffer buffer
+                                (-let* (((&hash "range") link)
+                                        (start (lsp--position-to-point (gethash "start" range)))
+                                        (end (lsp--position-to-point (gethash "end" range)))
+                                        (button (make-button start end 'action
+                                                             (lsp--document-link-keymap link))))
+                                  (push button lsp--link-overlays))))
+                            links))))
+    (cancel-timer lsp--links-idle-timer)
+    (setq-local lsp--links-idle-timer nil)))
+
+(defun lsp--document-link-handle-target (url)
+  (let* ((parsed-url (url-generic-parse-url (url-unhex-string url)))
+         (type (url-type parsed-url))
+         (file (decode-coding-string (url-filename parsed-url)
+                                     locale-coding-system)))
+    (if (string= type "file")
```

Comment:
We support multiple protocol schema in addition to `file:`. For example, jdt ls supports `jdt:` and `chelib:` - take a look at `lsp--uri-to-path`. I suspect that if you check for http/https and then use browse-url and fallback to lsp--uri-to-path in case it is not `http`/`https` will cover more cases.